src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -3191,11 +3185,19 @@ public void abort(Executor executor) throws SQLException {
                 SQLServerException.makeFromDriverError(this, this, form.format(msgArgs), null, true);
             }
         }
+        if (null == executor) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidArgument"));
+            Object[] msgArgs = {"executor"};
+            SQLServerException.makeFromDriverError(null, null, form.format(msgArgs), null, false);
+        } else {
 
-        setState(State.Closed);
+            // Always report the connection as closed for any further use, no matter
+            // what happens when we try to clean up the physical resources associated
+            // with the connection using executor.
```

Comment:
/*
*/
and, might as well change line 3178 to null != executor?

Reply:
Done